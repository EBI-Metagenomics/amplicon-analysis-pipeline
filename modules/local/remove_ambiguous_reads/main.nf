
process REMOVE_AMBIGUOUS_READS {
    // Run DADA2 pipeline including read-tracking

    label 'light'

    input:
    tuple val(meta), val(var_region), path(reads)

    output:
    tuple val(meta), val(var_region), path("*noambig*fastq.gz"), emit: noambig_out

    """
    if [[ ${meta.single_end} = true ]]; then
        python /hps/software/users/rdf/metagenomics/service-team/users/chrisata/asv_gen/bin/remove_ambiguous_reads.py -f $reads -s ${meta.id}
    else 
        python /hps/software/users/rdf/metagenomics/service-team/users/chrisata/asv_gen/bin/remove_ambiguous_reads.py -f ${reads[0]} -r ${reads[1]} -s ${meta.id}
    fi
    
    """

}
