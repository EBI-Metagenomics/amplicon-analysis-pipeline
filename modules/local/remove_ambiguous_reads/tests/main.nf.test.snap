{
    "noambig_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "16S-V3-V4",
                    [
                        "test_noambig_1.fastq.gz:md5,34834a5a828a6feb102330d4d1e2e135",
                        "test_noambig_2.fastq.gz:md5,ff8ed43509b9278821dd47e6cf17fbd4"
                    ]
                ]
            ]
        ],
        "timestamp": "2023-11-06T18:19:51.786528168"
    }
}